.buildkite/pipeline.yml

```diff
@@ -107,3 +107,69 @@ steps:
         depends_on:
           - "operator-image-build"
           - "e2e-tests-image-build"
+
+  - group: redhat-release
+
+    steps:
+
+      - label: "build operatorhub tool"
+        key: "build-operatorhub-tool"
+        if: build.message == "run operatorhub release"
+        commands:
+           - cd hack/operatorhub
+           - make build
+           - buildkite-agent artifact upload "bin/operator*"
+
+      - label: "Generate CRDs"
+        key: "generate-crds"
+        if: build.message == "run operatorhub release"
+        depends_on:
+          - "build-operatorhub-tool"
+        commands:
+           - git checkout $$OHUB_BRANCH
+           - make generate-crds-v1
+        artifact_paths:
+        - "config/*.yaml"
+
+      - label: "run container push"
+        key: "redhat-container-push"
+        if: build.message == "run operatorhub release"
+        depends_on:
+          - "generate-crds"
+        commands:
+           - buildkite-agent artifact download "bin/operator*" /usr/local/
+           - chmod u+x /usr/local/bin/operatorhub
+           - /usr/local/bin/operatorhub container push --dry-run=false
+
+      - label: "run container preflight"
+        key: "redhat-preflight"
+        if: build.message == "run operatorhub release"
+        depends_on:
+          - "redhat-container-push"
+        commands:
+           - export OHUB_API_KEY=$$(vault read -field=api-key secret/ci/elastic-cloud-on-k8s/operatorhub-release-redhat)
+           - export OHUB_PROJECT_ID=$$(vault read -field=project-id secret/ci/elastic-cloud-on-k8s/operatorhub-release-redhat)
+           - "echo curl -s -G \"https://catalog.redhat.com/api/containers/v1/projects/certification/id/$$OHUB_PROJECT_ID/images?filter=repositories.tags.name==$$OHUB_TAG\" -H \"X-API-KEY: $$OHUB_API_KEY\" > /tmp/redhat.json"
+           - "curl -s -G \"https://catalog.redhat.com/api/containers/v1/projects/certification/id/$$OHUB_PROJECT_ID/images?filter=repositories.tags.name==$$OHUB_TAG\" -H \"X-API-KEY: $$OHUB_API_KEY\" > /tmp/redhat.json"
+           - cat /tmp/redhat.json
+           # The following command will fail if no data was returned, so preflight will not run if it's already run
+           - jq -e '.data[0]' /tmp/redhat.json && exit 0
+           - curl -L -o /tmp/preflight https://github.com/redhat-openshift-ecosystem/openshift-preflight/releases/download/1.2.1/preflight-linux-amd64
+           - chmod u+x /tmp/preflight
+           - vault read -format=json -field=data secret/ci/elastic-cloud-on-k8s/operatorhub-release-preflight > /tmp/auth.json
+           - /tmp/preflight check container quay.io/redhat-isv-containers/$$OHUB_PROJECT_ID:$$OHUB_TAG --pyxis-api-token=$$OHUB_API_KEY --certification-project-id=$$OHUB_PROJECT_ID --submit -d /tmp/auth.json
```

Comment:
When there are so many commands, I think it will be more appropriate to write a bash script (`.buildkite/scripts/release/operatorhub-preflight.sh`). This would avoid to think about the $$ for example.

Did you consider to move this in the tool? I feel like `/usr/local/bin/operatorhub container preflight` would have been nice. I'm good with this version too.

Reply:
I'll get this contained within a script.  👍 

> Did you consider to move this in the tool? I feel like `/usr/local/bin/operatorhub container preflight` would have been nice. I'm good with this version too.

Oh yes I tried to get this fully into this tool, see https://github.com/redhat-openshift-ecosystem/openshift-preflight/issues/845